topo/node/ixia/ixia.go

```diff
@@ -208,29 +190,21 @@ func (n *Node) Create(ctx context.Context) error {
 	log.Infof("Creating deployment for node resource %s", n.Name())
 	desiredState := "DEPLOYED"
 
-	crd, err := n.getCRD(ctx)
+	c, err := ixclient.NewForConfig(n.RestConfig)
 	if err != nil {
 		return err
 	}
-	crd.Spec.DesiredState = desiredState
-	crd.Status = ixiatg.IxiaTGStatus{}
 
-	crdBytes, err := json.Marshal(crd)
+	unStrCRD, err := c.IxiaTG(n.Namespace).Unstructured(ctx, n.Name(), metav1.GetOptions{})
 	if err != nil {
-		return fmt.Errorf("could not marshal ixia CRD to JSON: %v", err)
+		return err
 	}
 
-	log.Infof("Updating ixia CRD (desiredState=%s) ...", desiredState)
+	crdSpec := unStrCRD.UnstructuredContent()["spec"].(map[string]interface{})
+	crdSpec["desired_state"] = desiredState
 
-	err = n.KubeClient.CoreV1().RESTClient().
-		Patch(types.MergePatchType).
-		AbsPath("/apis/network.keysight.com/v1beta1").
-		Namespace(n.Namespace).
-		Resource(ixiaResource).
-		Name(n.Name()).
-		Body(crdBytes).
-		Do(ctx).
-		Error()
+	log.Infof("Updating ixia CRD (desiredState=%s) ...", desiredState)
+	_, err = c.IxiaTG(n.Namespace).Update(ctx, unStrCRD, metav1.UpdateOptions{})
```

Comment:
can you add a comment here on why update is being used instead of create?

Reply:
Added comment for TopologySpecs and Create operations.